compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -199,6 +199,11 @@ pub unsafe fn create_module(
         }
     }
 
+    if sess.relocation_model() == RelocModel::Pie {
+        llvm::LLVMRustSetModulePICLevel(llmod);
+        llvm::LLVMRustSetModulePIELevel(llmod);
+    }
```

Comment:
Nit: could you merge this with the block above?
```rust
    let reloc_model = sess.relocation_model();
    if matches!(reloc_model, RelocModel::Pic | RelocModel::Pie) {
        llvm::LLVMRustSetModulePICLevel(llmod);
        // PIE is potentially more effective than PIC, but can only be used in executables.
        // If all our outputs are executables, then we can relax PIC to PIE.
        if reloc_model == RelocModel::Pie || sess.crate_types().iter().all(|ty| *ty == CrateType::Executable) {
            llvm::LLVMRustSetModulePIELevel(llmod);
        }
    }
```

Reply:
Done.